src/Nethermind/Nethermind.Config/Enode.cs

```diff
@@ -36,16 +36,30 @@ public Enode(PublicKey nodeKey, IPAddress hostIp, int port)
 
         public Enode(string enodeString)
         {
-            ArgumentException GetDnsException(string hostName, Exception innerException = null) => new($"{hostName} is not a proper IP address nor it can be resolved by DNS.", innerException);
+            ArgumentException GetDnsException(string hostName, Exception innerException = null) =>
+                new($"{hostName} is not a proper IP address nor it can be resolved by DNS.", innerException);
+
+            ArgumentException GetPortException(string hostName) =>
+                new($"Can't get Port for host {hostName}.");
 
             string[] enodeParts = enodeString.Split(':');
             string[] enodeParts2 = enodeParts[1].Split('@');
             _nodeKey = new PublicKey(enodeParts2[0].TrimStart('/'));
-            Port = int.Parse(enodeParts[2]);
-            var host = enodeParts2[1];
+            string host = enodeParts2[1];
+            if (enodeParts.Length > 2 && int.TryParse(enodeParts[2], out int port))
+            {
+                Port = port;
+            }
+            else
+            {
+                throw GetPortException(host);
+            }
+
```

Comment:
This, while better in terms of verbosity, don't change any meaningful in terms of how it works?